common/src/main/java/org/apache/drill/common/config/DrillConfig.java

```diff
@@ -138,34 +154,83 @@ public static DrillConfig create(Properties testConfigurations) {
     return create(null, testConfigurations, true);
   }
 
-  public static DrillConfig create(String overrideFileName, boolean enableServerConfigs) {
-    return create(overrideFileName, null, enableServerConfigs);
+  /**
+   * ...
+   * @param overrideFileResourcePathname
+   *          see {@link #create(String)}'s {@code overrideFileResourcePathname}
+   */
+  public static DrillConfig create(String overrideFileResourcePathname, boolean enableServerConfigs) {
+    return create(overrideFileResourcePathname, null, enableServerConfigs);
   }
 
-  private static DrillConfig create(String overrideFileName, Properties overriderProps, boolean enableServerConfigs) {
-    overrideFileName = overrideFileName == null ? CommonConstants.CONFIG_OVERRIDE : overrideFileName;
-
-    // first we load defaults.
+  /**
+   * ...
```

Comment:
I think if you should just put in explicit todo. However I'm not even sure that it makes sense to document these methods individually. There is a comment that is very descriptive about what all of these do, it's just above the wrong version of the method. If you move it above the one that takes all of the parameters, I think it would be fine to even leave out the parameter descriptions on the other versions and have a common message directing people to the one descriptive comment (which also describes all possible parameters). No need for a todo, not a bunch of work up front.